hpux/generate_wazuh_packages.sh

```diff
@@ -115,80 +103,151 @@ check_version(){
+  rm ${install_path}/wodles/oscap/content/*.xml
+  pkg_name="wazuh-agent-${wazuh_version}-${wazuh_revision}-hpux-11v3-ia64.tar"
+  tar cvpf ${target_dir}/${pkg_name} ${install_path} /etc/ossec-init.conf /sbin/init.d/wazuh-agent /sbin/rc2.d/S97wazuh-agent /sbin/rc3.d/S97wazuh-agent
+
+  if [ "${compute_checksums}" = "yes" ]; then
+    cd ${target_dir}
+    pkg_checksum="$(openssl dgst -sha512 ${pkg_name})"
+    echo "${pkg_checksum}  ${pkg_name}" > ${checksum_dir}/${pkg_name}.sha512
+  fi
 }
 
 #Uninstall agent.
 
 clean() {
-  $INSTALL/bin/ossec-control stop
-  rm -rf $INSTALL
+  exit_code=$1
+  ${install_path}/bin/ossec-control stop
+  rm -rf ${install_path}
   rm /etc/ossec-init.conf
   find /sbin -name "*wazuh-agent*" -exec rm {} \;
   userdel ossec
   groupdel ossec
+
+  exit ${exit_code}
 }
 
-show_help()
-{
-  echo "
-  This scripts build wazuh package for HPUX.
-  USAGE: Command line options available:
-    -h, --help       Displays this help.
-    -d, --download   Download Wazuh repository.
-    -b, --build      Builds HPUX package.
-    -u, --utils      Download and install utilities and dependencies.
-    -c, --clean-all  Clean sources and generated files.
-
-  USAGE EXAMPLE:
-  --------------
-    ./generate_wazuh_packages.sh [option] [branch_tag] [revision]
-    ./generate_wazuh_packages.sh -d branches/3.3 1
-  "
+show_help() {
+  echo
+  echo "Usage: $0 [OPTIONS]"
+  echo
+  echo "    -e Install all the packages necessaries to build the TAR package"
+  echo "    -b <branch> Select Git branch. Example v3.5.0"
+  echo "    -s <tar_directory> Directory to store the resulting tar package. By default: /tmp/build"
+  echo "    -p <tar_home> Installation path for the package. By default: /var"
+  echo "    -c, --checksum Compute the SHA512 checksum of the TAR package."
+  echo "    -h Shows this help"
+  echo
+  exit $1
+}
+
+build_package() {
+  download_source
+  compile
+  create_package
+}
+
+# Main function, processes user input
+main() {
+  # If the script is called without arguments
+  # show the help
+  if [[ -z $1 ]] ; then
+    show_help 0
+  fi
+
+  build_env="no"
+  build_pkg="no"
+
+  while [ -n "$1" ]
+  do
+    case $1 in
+      "-b")
+        if [ -n "$2" ]
+        then
+          wazuh_branch="$2"
+          build_pkg="yes"
+          shift 2
+        else
+          show_help 1
+        fi
+      ;;
+      "-h")
+        show_help
+        exit 0
+      ;;
+      "-e")
+        build_environment
+        exit 0
+      ;;
+      "-p")
+        if [ -n "$2" ]
+        then
+          install_path="$2"
+          shift 2
+        else
+          show_help 1
+        fi
+      ;;
+      "-s")
+        if [ -n "$2" ]
+        then
+          target_dir="$2"
+          shift 2
+        else
+          show_help 1
+        fi
+      ;;
+      "-c" | "--checksum")
+          if [ -n "$2" ]; then
+            checksum_dir="$2"
+            compute_checksums="yes"
+            shift 2
+          else
```

Comment:
This `else` do the same as initial `if`.

Reply:
Solved here: https://github.com/wazuh/wazuh-packages/commit/77debccdab0c5f437d22cac2ec98a747cd9b50a4